app/src/main/java/org/wikipedia/feed/suggestededits/SuggestedEditsCardView.kt

```diff
@@ -0,0 +1,147 @@
+        showImageOrExtract()
+    }
+
+    private fun showImageOrExtract() {
+        if (TextUtils.isEmpty(summary!!.thumbnailUrl)) {
+            viewArticleImage.visibility = View.GONE
+            viewArticleExtract.visibility = View.VISIBLE
+            divider.visibility = View.VISIBLE
+            viewArticleExtract.text = StringUtil.fromHtml(summary!!.extractHtml)
+            viewArticleExtract.maxLines = ARTICLE_EXTRACT_MAX_LINE_WITHOUT_IMAGE
+        } else {
+            viewArticleImage.visibility = View.VISIBLE
+            viewArticleExtract.visibility = View.GONE
+            divider.visibility = View.GONE
+            viewArticleImage.loadImage(Uri.parse(summary!!.thumbnailUrl))
+        }
+    }
+
+    private fun prepareViews() {
+        viewArticleContainer.minimumHeight = 0
+        viewArticleExtract.text = ""
+        viewArticleTitle.text = ""
+        callToActionText.text = ""
+        viewArticleImage.loadImage(null)
+        headerView.visibility = View.GONE
+        viewAddDescriptionButton.visibility = View.GONE
+        cardItemProgressBar.visibility = View.GONE
+        viewArticleSubtitleContainer.visibility = View.GONE
+        viewArticleExtract.visibility = View.GONE
+        divider.visibility = View.GONE
+        suggestedEditsItemRootView.setPadding(0, 0, 0, 0)
+        val param = cardView.layoutParams as FrameLayout.LayoutParams
+        param.setMargins(0, 0, 0, 0)
+        cardView.useCompatPadding = false
+        cardView.setContentPadding(0, 0, 0, 0)
+        cardView.setOnClickListener {
+            if (callback != null && card != null) {
+                callback!!.onSuggestedEditsCardClick(if (isTranslation) targetPageTitle!! else summary!!.getPageTitle(WikiSite.forLanguageCode(sourceLangCode)), sourceDescription, sourceLangCode, this)
+            }
+        }
+    }
+
+    override fun onDetachedFromWindow() {
+        disposables.clear()
+        super.onDetachedFromWindow()
+    }
+
+    private fun header(card: SuggestedEditsCard) {
+        headerView.visibility = View.VISIBLE
+        headerView!!.setTitle(card.title())
+                .setSubtitle(card.subtitle())
+                .setImage(R.drawable.ic_mode_edit_white_24dp)
+                .setImageCircleColor(ResourceUtil.getThemedAttributeId(context, R.attr.material_theme_de_emphasised_color))
+                .setLangCode(if (isTranslation) card.wikiSite().languageCode() else "")
+                .setCard(card)
+                .setCallback(callback)
+    }
+
+    fun showAddedDescriptionView(addedDescription: String?) {
+        if (!TextUtils.isEmpty(addedDescription)) {
+            viewArticleSubtitleContainer.visibility = View.VISIBLE
+            viewAddDescriptionButton.visibility = View.GONE
+            viewArticleSubtitleAddedBy.visibility = View.VISIBLE
+            viewArticleSubtitleEdit.visibility = View.VISIBLE
+            viewArticleSubtitle.text = addedDescription
```

Comment:
The `addedDescription` should be capitalized.

Reply:
Done